src/app/Controllers/ParameterValidator.cs

```diff
@@ -179,10 +319,38 @@ public static JsonResult MovieId(string movieId, string method, ILogger logger)
                 !int.TryParse(movieId.Substring(2), out int val) ||
                 val <= 0)
             {
-                return GetAndLogBadParam("Invalid Movie ID parameter", method, logger);
+                return GetAndLogInvalidMovieIdParameter(method, logger);
             }
 
             return null;
         }
+
+        /// <summary>
+        /// creates ErrorResponse and returns JsonResult with StatusCode for HTTP response
+        /// </summary>
+        /// <param name="logger"></param>
+        /// <param name="method"></param>
+        /// <param name="innerErrorType"></param>
+        /// <param name="message"></param>
+        /// <param name="httpStatusCode"></param>
+        /// <param name="errorCode"></param>
+        /// <param name="target"></param>
+        /// <param name="minValue"></param>
+        /// <param name="maxValue"></param>
+        /// <returns></returns>
+        private static JsonResult CreateAndFormatError(ILogger logger, string method, InnerErrorType innerErrorType, string message,
+            HttpStatusCode httpStatusCode, string errorCode, string target, object minValue, object maxValue)
+        {
+            logger.LogWarning($"InvalidParameter|{method}|{message}");
+
+            var httpInnerError = new InnerError(innerErrorType, minValue, maxValue);
```

Comment:
That is a lot of garbage collection to return an error string.  Based on what we saw in sentinel, I would use a format string or string builder to avoid the garbage collection, improve scale and reduce ddos vector

Reply:
@bartr if performance is the key driver, wouldn't a constant string achieve the best result? The approved PR in the Java repo uses this approach. Would this be acceptable for the C# repo too?

Reply:
perf is definitely a driver, but maintainability is as well. It seems to me having multiple full strings would be a lot of duplication which would make code maintenance more challenging. Seems a lot of that could be encapsulated in a format template but I haven't done a deep dive so maybe not. I'm OK either way, just think about developer experience and long-term code maintenance as well as perf / garbage collection.

Reply:
Updated to use StringBuilder